core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -814,6 +815,14 @@ private RelationPlan createTableExecutePlan(Analysis analysis, TableExecute stat
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, table.getName());
         TableExecuteHandle executeHandle = analysis.getTableExecuteHandle().orElseThrow();
 
+        if (!analysis.getTableExecuteHandle().orElseThrow().isReadsData()) {
```

Comment:
You have `analysis.getTableExecuteHandle().orElseThrow()` assigned to variable one line above